extensions/amp-selector/0.2/selector.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as CSS from './selector.css';
+import * as Preact from '../../../src/preact';
+import {useState} from '../../../src/preact';
+
+const SelectorContext = Preact.createContext({});
+
+/**
+ * @param {!JsonObject} props
+ * @return {PreactDef.Renderable}
+ */
+export function Selector(props) {
+  const {
+    'value': value,
+    'defaultValue': defaultValue,
+    'disabled': disabled,
+  } = props;
+  const isMultiple = props['multiple'] !== undefined;
```

Comment:
Multiple should go into `const {} = props` above. And it's already given us in the right way by React, so we don't need to do `!==undefined`. E.g. `<Selector multiple>` will pass `true`. We can instead do this:

```
const {
  ...
  'mutliple': multiple = false,
} = props;
```

Reply:
If it's not true now, we should make it true :). In this particular case we could add more stuff into `{attr: 'multiple'}`. I think we should `{attr: 'multiple', type: 'boolean'}` and use this type to correctly interpret the empty attribute or any other non-empty value.